google-factcheck-explorer-bot/README.md

```diff
@@ -0,0 +1,90 @@
+# Google factcheck explorer bot
+
+Shows Notes with Claim Review content that has been imported from https://toolbox.google.com/factcheck/explorer that is simlar to the Project Media item.
+
+# Overview
+
+NOTE: These instructions include references to documentation and AWS infrastructure that are not
+visable outside the Meedan organization. Please contact us if you need information about any of 
+these resources to install this code on your own system.
+
+## Background data
+For items to become availible to be displayed by the bot
+* ClaimReivew objects are parsed daily by Fetch plugin `fetch/lib/claim_review_parsers/google_fact_check.rb`
+    - TODO: replace this with more efficient ingest process, perhaps based on the code in `/ingest`
+* The google-fact-check-tools workspace https://checkmedia.org/google-fact-check-tools/project/15547 listens for 
+new ClaimReviews and stored in Check under its team id.  
+* google-fact-check-tools also publishes the content to a "shared feed' associated with its team id, with access permission determined by API key. 
+* The items on the shared feed are availible for similarity queries via Check API
+* NOTE: usually items in the project need to be in the 'published' state to be availible for similarity matching. It may be necessary to toggle them using a script like https://github.com/meedan/check-scripts/blob/main/publish_imported_reports.rb
+
+## Bot operation
+When this bot is configured in a workspace
+* The bot listens on a webhook for new ProjectMedia creation events for a team, configured as per https://meedan.atlassian.net/wiki/spaces/ENG/pages/1126268929/How+to+configure+a+webhook+for+a+Check+Bot
+```
+    bot_user = BotUser.where(name: "Google fact check workspace API Client")[0]
+    bot_user.set_request_url = "https://qyrh1ewsd0.execute-api.eu-west-1.amazonaws.com/default/qa-google-factcheck-explorer-bot"
+    bot_user.set_events = [{"event"=>"create_project_media", "graphql"=>"dbid, title, description, type"}]
+    bot_user.save!
+```
+* The text from the PM is similarity compared with the availible set of ClaimReview items via Check API in a query 
+managed by an AWS Lambda function defined in `/google-factcheck-explorer-bot-lambda`
+* Any resulting ClaimReview links are written back as 'comments' on the ProjectMedia items in the workspace to be displayed as Notes the sidebar
+* An appropriately configured API key is needed to give bot permissions to write to the workspace. This is done by mapping the BotUser to the team https://meedan.atlassian.net/wiki/spaces/ENG/pages/1126105091/How+to+create+an+API+key+for+a+Check+workspace
+
+# Bot testing setup
+## Testing the background side
+* Setup a workspace in QA to host the content (will this work or need to do locally)
+* Configure a shared feed for it and an api key https://meedan.atlassian.net/wiki/spaces/ENG/pages/1126105091
```

Comment:
Don't actually need a Shared Feed now. That was only something I did before we fixed the bug preventing searching a workspace with the API key